set.mm

```diff
@@ -369106,6 +369106,95 @@ varying definitions (some start from 0, others start from 1), but
+       This definition check relies on the database being very much like
+       set.mm, down to the names of certain constants and types, so it
+       cannot apply to all Metamath databases... but it is useful in set.mm.
+       In this definition check, a $a-statement with a given label and
+       typecode ` |- ` passes the test if and only if it
+       respects the following rules (these rules imply that we have
+       an unambiguous tree parse, which is also checked):
+       </p>
+
+       <ol>
+       <li>The expression must be a biconditional or an equality (i.e. its
+           root-symbol must be ` <-> ` or ` = `).
+           If the proposed definition passes this first rule, we then
+           define its definiendum as its left hand side (LHS) and
+           its definiens as its right hand side (RHS).
+           We define the *defined symbol* as the root-symbol of the LHS.
+           We define a *dummy variable* as a variable occurring
+           in the RHS but not in the LHS.
+           Note that the "root-symbol" is the root of the considered tree;
+           it need not correspond to a single token in the database
+           (e.g., see ~ w3o or ~ wsb ).
+       <li>The defined symbol should not occur in an earlier
+           $a-statement with typecode ` |- ` or in any another place
+           in this labelled expression.
+       <li>No two variables occurring in the LHS should share a
+           disjoint variable (DV) condition.
+           Without this rule, you can define something like
+           cbar $a wff Bar x y $.
+           ${ $d x y $. df-bar $a |- ( Bar x y <-> x = y ) $. $}
+           and now "Bar x x" is not eliminable;
+           there is no way to prove that it means anything in particular,
+           because the definitional theorem that is supposed to be
+           responsible for connecting it to the original language wants
+           nothing to do with this expression, even though it is well
+           formed.
+       <li>All dummy variables are required to be disjoint from any
+           other (dummy or not) variable occurring in this labelled expression.
+       <li>Either
+           (a) there must be no non-setvar dummy variables, or
+           (b) there must be a justification theorem.
+           The justification theorem must be of form
+           ` |- ( ` definiens root-symbol definiens' ` ) `
+           where definiens' is definiens but the dummy variables are all
+           replaced with other unused dummy variables of the same type.
+           Note that root-symbol is ` <-> ` or ` = ` , and that setvar
+           variables are simply variables with the ` setvar ` typecode.
+       <li>One of the following must be true:
+           (a) there must be no setvar dummy variables,
+           (b) there must be a justification theorem as described in rule 5, or
+           (c) we must prove that each dummy setvar variable is not free,
```

Comment:
Technically (c) subsumes (a)

Reply:
The modality here is wrong. This is a correctness condition, so you should drop "we must prove that". Probably we want to prove this, but the condition itself simply asserts that each dummy variable is not free.

Reply:
Okay, will tweak to clarify.